jwst/datamodels/model_base.py

```diff
@@ -306,6 +307,73 @@ def validate(self):
                               self._pass_invalid_values,
                               self._strict_validation)
 
+    def info(self):
+        """
+        Return datatype and dimension for each array or table
+        """
+        def get_field_info(path, instance):
+            field_info = []
+            if isinstance(instance, dict):
+                if path:
+                    path += '.'
+                for name, val in instance.items():
+                    new_path = path + name.lower()
+                    field_info.extend(get_field_info(new_path, val))
+            elif isinstance(instance, list):
+                for index, val in enumerate(instance):
+                    new_path = "%s[%d]" % (path, index)
+                    field_info.extend(get_field_info(new_path, val))
+            elif isinstance(instance, np.ndarray):
+                if instance.shape[0] > 0:
+                    shape_info = get_shape_info(instance)
+                    type_info = get_type_info(instance)
+                    field_info = [(path, shape_info, type_info)]
+            return field_info
+
+        def get_meta_info(meta):
+            meta_info = []
+            for attribute in ('filename', 'date', 'model_type'):
+                if hasattr(meta, attribute):
+                    value = getattr(meta, attribute)
+                    if value is not None:
+                        meta_info.append(attribute + ': ' + value)
+            return meta_info
+
+        def get_shape_info(instance):
+            if hasattr(instance, '_coldefs'):
+                nrows = instance.shape[0]
+                ncols = len(instance._coldefs)
+                shape_info = "{}R x {}C".format(nrows, ncols)
+            else:
+                # Display shape in fits order (reversed)
```

Comment:
I appreciate the desire for consistency with fits.info (and catfits), but I hope the ordering doesn't cause confusion with folks who are used to dealing with data model shapes in python order. I guess we'll find out.